File: backend/internal/domain/person.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

// Person represents a person registered in the system.
type Person struct {
	ID             uuid.UUID  `json:"id"`
	FullName       string     `json:"full_name"`
	BirthDate      *time.Time `json:"birth_date,omitempty"`
	DocumentType   string     `json:"document_type"`
	DocumentNumber *string    `json:"document_number,omitempty"`
	Gender         *string    `json:"gender,omitempty"`
	Email          *string    `json:"email,omitempty"`
	Phone          *string    `json:"phone,omitempty"`
	PhotoURL       *string    `json:"photo_url,omitempty"`
	ReferralSource *string    `json:"referral_source,omitempty"`
	Nationality    string     `json:"nationality"`
	CampusID       uuid.UUID  `json:"campus_id"`
	IsActive       bool       `json:"is_active"`
	CreatedAt      time.Time  `json:"created_at"`
	UpdatedAt      time.Time  `json:"updated_at"`
	CreatedBy      *uuid.UUID `json:"created_by,omitempty"`
}

// PersonDetail includes person data with addresses and roles.
type PersonDetail struct {
	Person
	Addresses []Address    `json:"addresses"`
	Roles     []PersonRole `json:"roles"`
}

// PersonListItem is a summary representation for list responses.
type PersonListItem struct {
	ID             uuid.UUID `json:"id"`
	FullName       string    `json:"full_name"`
	DocumentNumber *string   `json:"document_number,omitempty"`
	Phone          *string   `json:"phone,omitempty"`
	Roles          []string  `json:"roles"`
	IsActive       bool      `json:"is_active"`
}

// PersonListResult contains paginated person list data.
type PersonListResult struct {
	Data       []PersonListItem `json:"data"`
	Pagination Pagination       `json:"pagination"`
}

// Pagination holds pagination metadata.
type Pagination struct {
	Page       int `json:"page"`
	PerPage    int `json:"per_page"`
	Total      int `json:"total"`
	TotalPages int `json:"total_pages"`
}

// PersonFilter defines search/filter criteria for listing persons.
type PersonFilter struct {
	Query           string
	CampusID        uuid.UUID
	Page            int
	PerPage         int
	AgreementStatus string // "with_agreement", "without_agreement", "rejected"
}

// DuplicateMatch represents a potential duplicate person.
type DuplicateMatch struct {
	ID             uuid.UUID `json:"id"`
	FullName       string    `json:"full_name"`
	DocumentNumber string    `json:"document_number"`
	Campus         string    `json:"campus"`
	MatchType      string    `json:"match_type"`
}

// DuplicateCheckResult is the response for duplicate detection.
type DuplicateCheckResult struct {
	HasDuplicates bool             `json:"has_duplicates"`
	Matches       []DuplicateMatch `json:"matches"`
}

File: frontend/src/types/person.ts
export interface Person {
  id: string;
  full_name: string;
  birth_date?: string;
  document_type: string;
  document_number?: string;
  gender?: string;
  email?: string;
  phone?: string;
  photo_url?: string;
  referral_source?: string;
  nationality: string;
  campus_id: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  created_by?: string;
}

export interface PersonListItem {
  id: string;
  full_name: string;
  document_number?: string;
  phone?: string;
  roles: string[];
  is_active: boolean;
}

export interface Address {
  id: string;
  person_id: string;
  street?: string;
  number?: string;
  complement?: string;
  neighborhood?: string;
  city?: string;
  state?: string;
  zip_code?: string;
  country: string;
  is_primary: boolean;
  created_at: string;
  updated_at: string;
}

export interface PersonRole {
  id: string;
  person_id: string;
  role_type: string;
  professional_specialty?: string;
  is_active: boolean;
  activated_at: string;
  deactivated_at?: string;
  activated_by?: string;
  deactivated_by?: string;
  notes?: string;
}

export interface VolunteerAgreement {
  id: string;
  person_id: string;
  person_role_id: string;
  campus_id: string;
  status: 'PENDING' | 'ACCEPTED' | 'REJECTED';
  signature_method?: 'DIGITAL' | 'MANUAL_UPLOAD';
  accepted_at?: string;
  accepted_by_user?: string;
  ip_address?: string;
  user_agent?: string;
  document_path?: string;
  uploaded_at?: string;
  uploaded_by?: string;
  rejected_at?: string;
  rejection_reason?: string;
  agreement_version: string;
  notes?: string;
  created_at: string;
  updated_at: string;
}

export interface PersonDetail extends Person {
  addresses: Address[];
  roles: PersonRole[];
}

export interface Pagination {
  page: number;
  per_page: number;
  total: number;
  total_pages: number;
}

export interface PersonListResponse {
  data: PersonListItem[];
  pagination: Pagination;
}

export interface DuplicateMatch {
  id: string;
  full_name: string;
  document_number: string;
  campus: string;
  match_type: string;
}

export interface DuplicateCheckResult {
  has_duplicates: boolean;
  matches: DuplicateMatch[];
}

export interface HistoryEntry {
  type: string;
  id: string;
  date: string;
  summary: string;
  service_type?: string;
  professional?: string;
  status?: string;
  campaign?: string;
}

export interface CreatePersonInput {
  full_name: string;
  birth_date?: string;
  document_type: string;
  document_number?: string;
  nationality?: string;
  gender?: string;
  email?: string;
  phone?: string;
  referral_source?: string;
  address?: AddressInput;
  sync_id?: string;
}

export interface UpdatePersonInput {
  full_name: string;
  birth_date?: string;
  document_type: string;
  document_number?: string;
  nationality?: string;
  gender?: string;
  email?: string;
  phone?: string;
  referral_source?: string;
  address?: AddressInput;
}

export interface SelfRegisterInput {
  full_name: string;
  birth_date?: string;
  document_type: string;
  document_number?: string;
  nationality?: string;
  gender?: string;
  phone?: string;
  referral_source?: string;
  address?: AddressInput;
  role_type: 'VOLUNTEER' | 'ASSISTED';
  campus_id: string;
}

export interface AddressInput {
  street?: string;
  number?: string;
  complement?: string;
  neighborhood?: string;
  city?: string;
  state?: string;
  zip_code?: string;
  country?: string;
}

export interface AddRoleInput {
  role_type: string;
  professional_specialty?: string;
  notes?: string;
}
